<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FuzzingWindow</class>
 <widget class="QDialog" name="FuzzingWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1188</width>
    <height>701</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Fuzzing Window</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="1,1,3,0,0,6,0,0,0">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Interframe Timing (ms)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinTiming">
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="value">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Number of data bytes</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBytes">
         <property name="maximum">
          <number>64</number>
         </property>
         <property name="value">
          <number>8</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Burst Rate</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBurst">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>10</number>
         </property>
         <property name="value">
          <number>1</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Bus or Buses to Send On</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="cbBuses"/>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>ID Scanning</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QRadioButton" name="rbSequentialID">
          <property name="text">
           <string>Seqential</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbRandomID">
          <property name="text">
           <string>Random</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>ID Selection</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QRadioButton" name="rbRangeIDSel">
          <property name="text">
           <string>Range of IDs</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbFilterIDSel">
          <property name="text">
           <string>Filter List</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="title">
        <string>Bit Scanning</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QRadioButton" name="rbSequentialBits">
          <property name="text">
           <string>Sequential</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbSweep">
          <property name="text">
           <string>Sweep</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="rbRandomBits">
          <property name="text">
           <string>Random</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="title">
        <string>IDs</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Start ID:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="txtStartID"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_7">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>End ID:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="txtEndID"/>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QListWidget" name="listID"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <widget class="QPushButton" name="btnAllFilters">
           <property name="text">
            <string>All</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnNoFilters">
           <property name="text">
            <string>None</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="label_8">
     <property name="text">
      <string>Byte Values (Always in hexadecimal)</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_10">
     <item>
      <widget class="QLabel" name="label_9">
       <property name="font">
        <font>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>0</string>
       </property>
       <property name="textFormat">
        <enum>Qt::PlainText</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="txtByte0"/>
     </item>
     <item>
      <widget class="QLabel" name="label_10">
       <property name="font">
        <font>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>1</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="txtByte1"/>
     </item>
     <item>
      <widget class="QLabel" name="label_11">
       <property name="font">
        <font>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>2</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="txtByte2"/>
     </item>
     <item>
      <widget class="QLabel" name="label_12">
       <property name="font">
        <font>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>3</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="txtByte3"/>
     </item>
     <item>
      <widget class="QLabel" name="label_13">
       <property name="font">
        <font>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>4</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="txtByte4"/>
     </item>
     <item>
      <widget class="QLabel" name="label_14">
       <property name="font">
        <font>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>5</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="txtByte5"/>
     </item>
     <item>
      <widget class="QLabel" name="label_15">
       <property name="font">
        <font>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>6</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="txtByte6"/>
     </item>
     <item>
      <widget class="QLabel" name="label_16">
       <property name="font">
        <font>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>7</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="txtByte7"/>
     </item>
     <item>
      <widget class="QLabel" name="label_17">
       <property name="text">
        <string>8</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineEdit"/>
     </item>
     <item>
      <widget class="QLabel" name="label_18">
       <property name="text">
        <string>9</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineEdit_2"/>
     </item>
     <item>
      <widget class="QLabel" name="label_19">
       <property name="text">
        <string>10</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineEdit_3"/>
     </item>
     <item>
      <widget class="QLabel" name="label_20">
       <property name="text">
        <string>11</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineEdit_4"/>
     </item>
     <item>
      <widget class="QLabel" name="label_21">
       <property name="text">
        <string>12</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineEdit_5"/>
     </item>
     <item>
      <widget class="QLabel" name="label_22">
       <property name="text">
        <string>13</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineEdit_6"/>
     </item>
     <item>
      <widget class="QLabel" name="label_23">
       <property name="text">
        <string>14</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineEdit_7"/>
     </item>
     <item>
      <widget class="QLabel" name="label_24">
       <property name="text">
        <string>15</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineEdit_8"/>
     </item>
    </layout>
   </item>
   <item>
    <widget class="CANDataGrid" name="bitfield" native="true">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>200</height>
      </size>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_7">
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="text">
      <string>White = Bit always 0, Green = Fuzz Bit, Black = Bit always set, Gray = Past end of data bytes</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="btnStartStop">
     <property name="text">
      <string>Start Fuzzing</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="lblNumFrames">
     <property name="font">
      <font>
       <pointsize>11</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string># of sent frames:</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CANDataGrid</class>
   <extends>QWidget</extends>
   <header>candatagrid.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>spinTiming</tabstop>
  <tabstop>spinBurst</tabstop>
  <tabstop>spinBytes</tabstop>
  <tabstop>cbBuses</tabstop>
  <tabstop>rbSequentialID</tabstop>
  <tabstop>rbRandomID</tabstop>
  <tabstop>rbRangeIDSel</tabstop>
  <tabstop>rbFilterIDSel</tabstop>
  <tabstop>rbSequentialBits</tabstop>
  <tabstop>rbSweep</tabstop>
  <tabstop>rbRandomBits</tabstop>
  <tabstop>txtStartID</tabstop>
  <tabstop>txtEndID</tabstop>
  <tabstop>listID</tabstop>
  <tabstop>btnAllFilters</tabstop>
  <tabstop>btnNoFilters</tabstop>
  <tabstop>btnStartStop</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
